mcs/class/monodoc/Monodoc.Ecma/EcmaDesc.cs

```diff
@@ -232,20 +232,27 @@ void ConstructCRef (StringBuilder sb)
 			if (DescKind == Kind.Namespace)
 				return;
 
-			sb.Append ('.');
+			if (!string.IsNullOrEmpty (Namespace))
+				sb.Append ('.');
+
 			sb.Append (TypeName);
 			if (GenericTypeArguments != null) {
-				sb.Append ('<');
-				int i=0;
-				foreach (var t in GenericTypeArguments) {
-					if (i > 0) {
-						sb.Append (",");
+				if (GenericTypeArguments.Any (a => a == null)) {
```

Comment:
I didn't notice that property ... so I will definitely refactor to use it (along with the other few comments in this area), thanks for pointing it out.

As for the intent behind this ... it's because when building up the ecma URLs, if the "type" is available, we can print out URLs such as SomeClass<T,K> ... but if even one of those type arguments is null, then we have no way of printing out the correct signature, and so it needs to default back to the SomeClass`2 form of the signature.